### ISSUE TITLE
[feature request] Increase user-friendliness of dataset.random_split

### ISSUE THREAD

**Issue by romanovacca** [pytorch/pytorch#11980]
## Issue description

Currently, when using the [random_split function](https://github.com/pytorch/pytorch/blob/master/torch/utils/data/dataset.py)  the parameters that need to be given are:
- dataset
- list that contains the lengths of splits to be produced

This means a user has to calculate these upfront and add them to the function as parameters like this:
`train_size = int(0.8 * len(dataset))`
`test_size = len(dataset) - train_size`
`train_dataset, test_dataset = torch.utils.data.random_split(dataset, [train_size,test_size])`

Wouldn't it be better if the second parameter was a list that contains the percentages in which the users wants the split to happen, like this:
`train_dataset, test_dataset = torch.utils.data.random_split(dataset,[0.8,0.2])` .

The result of this change would be cleaner code for the user and also I believer in a more natural way of creating the splits.

I would like to pick this up as a pull request!

**Comment by zou3519** [comment]
@romanovacca this sounds good. Please feel free to pick it up

**Comment by romanovacca** [comment]
I have created a solution. This is my first PR and I looked at the 'contributing' document, but don't know how I can unit test my changes using any existing tests. What I did now was locally run this and tested some scenarios and that went well. So is there anything else I can do before creating a PR?

**Comment by zou3519** [comment]
That's a good point, I'm not sure how to test randomness. Feel free to put up a PR and I'll take a look at it.

**Comment by jeffreyksmithjr** [comment]
NB on the analogous functionality from Spark: https://spark.apache.org/docs/2.3.1/api/scala/index.html#org.apache.spark.rdd.RDD@randomSplit(weights:Array[Double],seed:Long):Array[org.apache.spark.rdd.RDD[T]]

Just an example of an analogous API and implementation. In particular, I would highlight that the Spark implementation makes the choice of normalizing weights which don't sum to zero. An alternative for this normalization of weights functionality could be to simply throw an error, depending on the workflow desired.

**Comment by romanovacca** [comment]
@zou3519 PR is here: [#12068](https://github.com/pytorch/pytorch/pull/12068) . This is my first PR, so if I did it wrong let me know!

@jeffreyksmithjr I do not understand what you mean, could you explain it?

**Comment by ezyang** [comment]
FWIW the PR got NACKed by apaszke in review, so I don't know if this request is dead or if there's another way to do it that resolves his concern.
